src/scipp/io/hdf5.py

```diff
@@ -431,7 +431,13 @@ def read(group: h5.Group):
 
 
 def _direct_io(cls, convert=None):
-    type_name = cls.__name__
+    # Use fully qualified name for numpy.bool to avoid confusion with builtin bool."
+    # Numpy bools have different names in 1.x and 2.x
+    # TODO: This should be fixed once we drop support for numpy<2
+    if cls.__module__ == 'numpy' and cls.__name__[:4] == 'bool':
+        type_name = 'numpy.bool'
+    else:
+        type_name = cls.__name__
```

Comment:
Does this break compatibility with old files? If so, please add a script to https://github.com/scipp/scipp/tree/main/tools/migration